model/change_point_detection.go

```diff
@@ -369,10 +369,21 @@ func ReplaceChangePoints(ctx context.Context, env cedar.Environment, performance
 	return catcher.Resolve()
 }
 
-func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar.Environment, projectId string, pageSize int) (int, error) {
-	countKey := "count"
-	pipe := appendAfterBaseGetChangePointsByVersionAgg(projectId, bson.M{
-		"$count": countKey,
+type GetChangePointsGroupedByVersionOpts struct {
+	ProjectId        string
+	Page             int
+	PageSize         int
+	VariantRegex     string
+	VersionRegex     string
+	TaskRegex        string
+	TestRegex        string
+	MeasurementRegex string
+	ThreadLevels     []int
```

Comment:
I'm a bit confused how thread_level gets here: thread level is a higher-order construct that is specific to (some) server tests, and isn't a first-class attribute of the perf system in Cedar: there's a "test options" map in the info section, and I presume we're just mapping against that, but it makes it unclear what happens if we want to handle other kinds of arguments to tests. 

Maybe making this a map[string][]int named Arguments or Parameters would make more sense here.

Reply:
How it got here is simple: consumer wants to filter/sort by thread level! So I'd request that the customer facing API support `?thread_levels=2,4,16` for example, but how this is represented after parsing and in lower levels is up to you. Let me know what your prefer!

Reply:
Right but, tests and users can specify arbirary string-to-int (slice) mapping and the back-end doesn't need to know about them internally.

Reply:
i.e. `?arguments=<insert json here>`?

Reply:
Is it viable to just parse unknown query strings as things that go into arguments? 

json encoding query parameters seems... fun?

Reply:
Tix to handle this stuff since ui also needs an update https://jira.mongodb.org/browse/MAKE-1299